notificationdemo/source/java/src/org/kaaproject/kaa/demo/notification/NotificationDemo.java

```diff
@@ -51,74 +58,107 @@ public static void main(String[] args) {
-        @Override
-        public void onListUpdated(List<Topic> list) {
-            LOG.info("Topic list was updated");
-            showTopicList(list);
-            try {
-                //Try to subscribe to all new optional topics, if any.
-                List<Long> optionalTopics = extractOptionalTopicIds(list);
-                for(Long optionalTopicId : optionalTopics){
-                    LOG.info("Subscribing to optional topic {}", optionalTopicId);
-                }
-                kaaClient.subscribeToTopics(optionalTopics, true);
-            } catch (UnavailableTopicException e) {
-                LOG.error("Topic is unavailable, can't subscribe: {}", e.getMessage());
-            }
+    private static void showTopics() {
+        if (topics == null || topics.isEmpty()) {
+            LOG.info("Topic list is empty");
+            return;
+        }
+
+        LOG.info("Available topics:");
+        for (Topic topic : topics) {
+            LOG.info("Topic id: {}, name: {}, type: {}", topic.getId(), topic.getName(), topic.getSubscriptionType());
+        }
+
+        LOG.info("Subscribed on topics:");
+        for (Topic t : getOneTypeTopics(SubscriptionType.MANDATORY_SUBSCRIPTION)) {
+            LOG.info("Topic id: {}, name: {}, type: {}", t.getId(), t.getName(), t.getSubscriptionType().name());
         }
     }
 
-    private static List<Long> extractOptionalTopicIds(List<Topic> list) {
-        List<Long> topicIds = new ArrayList<>();
-        for (Topic t : list) {
-            if (t.getSubscriptionType() == SubscriptionType.OPTIONAL_SUBSCRIPTION) {
-                topicIds.add(t.getId());
+    private static List<Topic> getOneTypeTopics(SubscriptionType type) {
+        List<Topic> topics = new ArrayList<>();
+        for (Topic t : NotificationDemo.topics) {
+            if (t.getSubscriptionType() == type) {
+                topics.add(t);
             }
         }
-        return topicIds;
+        return topics;
     }
 
-    private static void showTopicList(List<Topic> topics) {
-        if (topics == null || topics.isEmpty()) {
-            LOG.info("Topic list is empty");
-        } else {
-            for (Topic topic : topics) {
-                LOG.info("Topic id: {}, name: {}, type: {}", topic.getId(), topic.getName(), topic.getSubscriptionType());
+    private static void subscribeTopic(long topicId) {
+        try {
+            kaaClient.subscribeToTopic(topicId, true);
+        } catch (UnavailableTopicException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private static Topic getTopic(long id) {
+        for (Topic t : topics)
+            if (t.getId() == id)
+                return t;
+        return null;
+    }
+
+    private static void unsubscribeOptionalTopics() {
+        List<Topic> topics = getOneTypeTopics(SubscriptionType.OPTIONAL_SUBSCRIPTION);
+        for (Topic t : topics) {
+            try {
+                kaaClient.unsubscribeFromTopic(t.getId());
```

Comment:
You can perform something like this: 

```
   kaaClient.unsubscribeFromTopics(getOneTypeTopics(SubscriptionType.OPTIONAL_SUBSCRIPTION), true);
```

see: http://docs.kaaproject.org/display/KAA/Using+notifications#Usingnotifications-Subscribetooptionaltopics